Zabbix 3.2

thespain opened the pull request:
Hello, Please see the commits in this PR to move this module from using Zabbix 2.4 to 3.2.

In addition to the changes to support Zabbix 3.2 I also modified the zabbix_agent.conf.erb template to remove some of the indentations. I found out that having a space in front of `PidFile` caused operating systems that use SysVinit to fail to stop and get the status of the zabbix_agentd service. For the single value variables I added a `-` to the opening tag, however that did not suffice for the variables containing arrays, as it would only not indent the first value in the array.

[I used the latest ERB language guide as a reference.](https://docs.puppet.com/puppet/4.9/lang_template_erb.html)

Please let me know what you think and if you have any comments or suggestions!

thespain commented:
@genebean I just realized that I haven't updated the repos for Suse... I'll get those corrected as soon as I can.

genebean commented:
This is awesome @thespain! One question: would you be willing to adjust your approach just a little so that we can make it work with more than one version of Zabbix? What I had in mind was some conditional logic so that places that have a Zabbix version number (like in templates) would fill that with a parameter instead of being hard coded. I'd like to see something like a `version` param that validated against a list of our supported versions. Here's kinda what I have in mind:

```puppet
# params.pp
$version = '3.2'

# init.pp
#param list:
$version = $::zabbixagent::params::version,

# below the params in the code section:
if ! ( $version in [ '2.4', '3.2', ] {
  fail("Zabbix ${version} is not supported but PR's are welcome.")
}
```

This list would include what the module supports before this PR and would also include your new version. In things like templates you could check this version param and not only use it to fill in values but also only include things if they belong there. Here's an example of each:

- repository paths could use this value as part of the string that makes up the URL
- the conf file could only include the new 3.2 settings if the user is using that version
- the conf file could only include things that were eliminated from the config if the user is still on the older version (not sure if that has happened or not)

If this is too much to ask just let me know. Thanks!

genebean commented on templates/zabbix.repo.erb:
```diff
@@ -14,14 +14,30 @@
+baseurl=http://repo.zabbix.com/zabbix/<%= @version %>/rhel/<%= @operatingsystemmajrelease %>/$basearch/
 enabled=1
 gpgcheck=1
+<% if @version == '2.4' -%>
```
This looks like it should be an if else if block instead of multiple if's

thespain replied:
Thanks, took care of this one.

genebean commented on templates/zabbix.repo.erb:
```diff
@@ -14,14 +14,30 @@
 baseurl=http://repo.zabbix.com/non-supported/rhel/<%= @operatingsystemmajrelease %>/$basearch/
 enabled=1
 gpgcheck=1
+<% if @version == '2.4' -%>
```
Same here

thespain replied:
And this one.

thespain commented:
@genebean Thanks for the suggestions and help on improvement!

genebean commented:
I just realized that with the addition of selectable version we should probably test that things work right when v2.4 is selected. If you're game for adding that, awesome. If not, no biggie, I'll add it before releasing.

thespain commented:
@genebean I finished the changes requested and squashed my commits. I also added a couple of tests for Ubuntu just to be safe. 

Just an reminder of a notable change: I modified the Suse repos because even though the original was more "official", it did not contain any packages for Zabbix later than v2.2. Due to that change, I ensured that the original `zabbix-agent` package was removed before installing the new version.

Please let me know if you would like to see any other changes.

Thanks
